### PR TITLE
[CI] Prepare the maui-template to work out of the repo pipeline.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Few things that needed to be added to run the tests in the megapipeline:

1. Allow provisionator to use a diff checkout directory (the megapipeline has several checkout repos).
2. Allow to add extra prepare steps that will be used to do a all the prapre steps needed to get the agent ready with the latests build.

### CODE REVIEW

**Review by PureWeen — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Comment by mattleibow** [comment]
@rmarinho any idea why the device tests pipeline starts automatically, but MAUI-public does not?

**Comment by mattleibow** [comment]
image

File: eng/pipelines/common/maui-templates.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ parameters:
     type: string
     default: ''
 
+  - name: checkoutDirectory
+    type: string
+    value: $(System.DefaultWorkingDirectory)
```
Comment (mandel-macaque) [inline eng/pipelines/common/maui-templates.yml:19]:
```suggestion
    default: $(System.DefaultWorkingDirectory)
```

**Review by mattleibow — APPROVED** [review]

**Pull request merged**